addons/mail/static/src/models/thread/thread.js

```diff
@@ -1336,6 +1325,20 @@ function factory(dependencies) {
             isCausal: true,
         }),
         channel_type: attr(),
+        /**
+         * States the `mail.chat_window` related to `this`. Serves as compute
+         * dependency. It is computed from the inverse relation and it should
+         * otherwise be considered read-only.
+         */
+        chatWindow: one2one('mail.chat_window', {
+            inverse: 'thread',
+        }),
```

Comment:
this should be a `one2many`. `mail.thread` is a uniquely identified thread in logic, while `mail.chat_window` maps a UI element. Could have several chat windows with same thread (not supported yet, but it feels wrong to put such a constraint in the modeling)

Reply:
It must fundamentally be a `one2one` because this is how it is stored on the server. There can only be one fold state per thread. Rest of the code is working based on this, especially the related `chatWindowIsFolded`.

Reply:
I would have preferred still allowing multiple chat windows for a single thread in modeling, but having source of truth of fold state in thread, so that all chat windows auto-sync when one of them is folded.

This looks dumb, and not actually used in practice... But IMO it puts much less constraint on design, so future changes become easier. This is especially important when we are gonna improve notions of "purely logical" records (e.g. `mail.thread`) and "UI" records (e.g. `mail.chat_window`). Conceptually a chat window is just a UI container of a thread.

Reply:
If there are multiple chat windows for a thread, it's impossible to decide the state of which chat window to store on the server. If we receive a fold or a close from the server, it's impossible to know which chat window to apply it to, etc.

It must be `one2one` for now, the whole sync logic is relying on it.

Reply:
Even open is relying it on it, calling open on thread already in chat window will do nothing, not open a new one.

Reply:
Ok, that's hopefully something to improve later on.